extensions/wikia/UserProfilePageV3/maintenance/migrateAvatarsViaService.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * Script migrates the user avatars from DFS to user avatars service
+ *
+ * @see PLATFORM-1419
+ *
+ * @author Macbre
+ * @ingroup Maintenance
+ */
+
+putenv( 'SERVER_ID=177' ); // run in the context of a community wiki
+
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+
+class AvatarsMigratorException extends Exception {}
+
+/**
+ * Maintenance script class
+ */
+class AvatarsMigrator extends Maintenance {
+
+	private $isDryRun = true;
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption('dry-run', 'Don\'t perform any operations [default]');
+		$this->addOption( 'force', 'Apply the changes made by the script' );
+
+		$this->mDescription = 'This script migrates the user avatars from DFS to user avatars service';
+	}
+
+	public function execute() {
+		// read options
+		$this->isDryRun = $this->hasOption( 'dry-run' ) || !$this->hasOption( 'force' );
+
+		if ($this->isDryRun) $this->output( "Running in dry-run mode!\n\n" );
+
+		$this->output( "Getting the list of all accounts...\n" );
+
+		// get all accounts
+		global $wgExternalSharedDB;
+		$db = $this->getDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$res = $db->select(
+			'`user`',
+			'user_id AS id',
+			[],
+			__METHOD__
+		);
```

Comment:
@jsutterfield, I've reverted f7716d3d8e183be85e63b309ab3303719bd7fe29

The plan is to first enable avatar uploads via the new service and migrate the avatars when we verify that the avatars service is stable. Hence, we can not query `user_preferences` table as it can have settings that are out of date (the preferences service uses a separate database to store settings) - get the list of **all** users and query the preferences service (via `User::getGlobalAttribute` method). This will make the process take longer, but "better safe than sorry" :)